doc/guides/west/repo-tool.rst

```diff
@@ -1,79 +1,289 @@
+the ``-m`` switch, the initial revision to check out using ``--mr``, and
+the location of the manifest file within the repository using ``--mf``.
+
+For example, running:
+
+.. code-block:: shell
+
+   west init -m https://github.com/zephyrproject-rtos/zephyr --mr v1.14.0 zp
+
+would clone the upstream official zephyr repository into :file:`zp/zephyr`,
+and check out the ``v1.14.0`` release. This command creates
+:file:`zp/.west`, and set the ``manifest.path`` :ref:`configuration option
+<west-config>` to ``zephyr`` to record the location of the manifest
+repository in the workspace. The default manifest file location is used.
+
+The ``-m`` option defaults to
+``https://github.com/zephyrproject-rtos/zephyr``. The ``--mr`` option
+defaults to ``master``. The ``--mf`` option defaults to ``west.yml``. If no
+``directory`` is given, the current working directory is used.
+
+**Option 2**: to create a workspace around an existing local manifest
+repository, use:
+
+.. code-block:: none
+
+   west init -l [--mf FILE] directory
+
+This creates :file:`.west` **next to** :file:`directory` in the file
+system, and sets ``manifest.path`` to ``directory``.
+
+As above, ``--mf`` defaults to ``west.yml``.
+
+The ``west init`` command does not clone any of the projects defined in the
+manifest file, regardless of whether ``-l`` is given. To do that, use
+``west update``.
+
+.. _west-update:
+
+``west update``
+---------------
+
+This command clones and updates the projects specified in the :term:`west
+manifest` file.
+
+.. code-block:: none
+
+   west update [-h] [--stats] [-f {always,smart}] [-k] [-r] [PROJECT ...]
+
+By default, this command:
+
+#. Parses the manifest file, usually :file:`west.yml`
+#. Clones any project repositories that are not already present locally
+#. Fetches any project revisions which are not already pulled from the
+   remote
+#. Sets each project's :ref:`manifest-rev <west-manifest-rev>` branch to the
+   revision specified for that project in the manifest file
+#. Checks out each ``manifest-rev`` in local working trees, as `detached
+   HEADs <https://git-scm.com/docs/git-checkout#_detached_head>`_
+
+To operate on a subset of projects only, specify them using the ``PROJECT``
+positional arguments, which can be either project names as given in the
+manifest file, or paths to the local project clones.
+
+To force this command to fetch from project remotes even if the revisions
+appear to be available locally, either use ``--fetch always`` or set the
+``update.fetch`` :ref:`configuration option <west-config>` to ``"always"``.
+
+For safety, ``west update`` uses ``git checkout --detach`` to check out a
+detached ``HEAD`` at the manifest revision for each updated project,
+leaving behind any branches which were already checked out. This is
+typically a safe operation that will not modify any of your local branches.
+
+If you would rather rebase any locally checked out branches instead, use
+the ``-r`` (``--rebase``) option.
```

Comment:
> Sorry I forgot what happens with detached HEADs in this case: `--rebase` (silently?) ignored?

That's deliberately undefined behavior for now because I think we may want to change it at some point. The current implementation just ignores. See https://github.com/zephyrproject-rtos/west/issues/381 for our existing discussion on the subject.

Reply:
Ha! I tend to forget "undefined" a lot, that explains.

Still, it's very scary how much @marc-hb commented on https://github.com/zephyrproject-rtos/west/issues/381 just a year ago and forgot all about it :-(

While waiting for 381 I think the docs should mention this as "explicitly undefined" because I suspect committing on detached HEADs is relatively common for the simple reason that it is "what's there".